### PR TITLE
Issue 395 Allowed Cross Domain Hosts

### PR DESCRIPTION
**Pull request by kingthorin**
At HIGH threshold check if URLs for included scripts fall within the same
context as the message being analyzed. Also update help entry and
ZapAddOn.xml.

https://github.com/zaproxy/zap-core-help/wiki/HelpUiDialogsOptionsPscanrules#threshold

Fixes zaproxy/zaproxy#395

### CODE REVIEW

File: src/org/zaproxy/zap/extension/pscanrules/CrossDomainScriptInclusionScanner.java [hunk 1]
```diff
@@ -116,7 +119,18 @@ private boolean isScriptFromOtherDomain (String host, String scriptURL){
 			String scriptHost = scriptURI.getHost();
 			if(scriptHost != null && !scriptHost.toLowerCase().equals(host.toLowerCase())){
 				result = true;
-			} 
+			}
+			if(Plugin.AlertThreshold.HIGH.equals(this.getLevel())) {
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/pscanrules/CrossDomainScriptInclusionScanner.java]: This if statement could also use the variable `result` (i.e. no need to check further if it's not a script from other domain).
(The variable could be renamed to `otherDomain`.)

File: src/org/zaproxy/zap/extension/pscanrules/CrossDomainScriptInclusionScanner.java [hunk 2]
```diff
@@ -116,7 +119,18 @@ private boolean isScriptFromOtherDomain (String host, String scriptURL){
 			String scriptHost = scriptURI.getHost();
 			if(scriptHost != null && !scriptHost.toLowerCase().equals(host.toLowerCase())){
 				result = true;
-			} 
+			}
+			if(Plugin.AlertThreshold.HIGH.equals(this.getLevel())) {
+				//Get a list of contexts that contain the original URL
+				List<Context> contextList=Model.getSingleton().getSession().getContextsForUrl(msg.getRequestHeader().getURI().toString());
+				for (Context context : contextList) {
+					if(context.isInContext(scriptURI.toString())) {
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/pscanrules/CrossDomainScriptInclusionScanner.java]: The conversion to String could be done just once, outside the for loop.

**Comment by thc202** [comment]
The commit message says "A HIGH..." should be "At HIGH..." instead?

**Comment by kingthorin** [comment]
Yup, will fixup shortly.

**Comment by thc202** [comment]
Looks good to me, thank you!

**Pull request merged**